client/client.go

```diff
@@ -468,6 +468,16 @@ func NewClient(options Options) (Client, error) {
 	return internal.NewClient(options)
 }
 
+// NewClientFromExisting creates a new client using the same connection as the
+// existing client. This means all options.ConnectionOptions are ignored and
+// options.HostPort is ignored. The existing client must have been created from
+// this package and cannot be wrapped. Currently, this always attempts an eager
+// connection even if the existing client was created with NewLazyClient and has
+// not made any calls yet.
+func NewClientFromExisting(existingClient Client, options Options) (Client, error) {
```

Comment:
It might make sense to make a narrower options type here with just the stuff that won't be ignored.

If that can be re-used in a backwards compat way inside the existing options, that's gravy. If not it might still be worth it just to make this less potentially confusing, but up to you.

Reply:
In theory I like splitting the struct into "connection options" and "other options" but I don't think it's worth it for this feature that most people won't ever use

Reply:
> It might make sense to make a narrower options type here with just the stuff that won't be ignored.

This would entail just copying all but two fields just for this method and hoping I properly keep them in sync when adding more. I think it's reasonable to reuse options explaining which are unused.